packages/astro/src/@types/astro-core.ts

```diff
@@ -139,8 +139,12 @@ export interface CollectionRSS {
 
 /** Generic interface for a component (Astro, Svelte, React, etc.) */
 export interface ComponentInstance {
+  $$metadata: {
+    modules: { module: ComponentInstance; specifier: string }[];
+    fileURL: URL;
+    metadataCache: Map<string, unknown>;
```

Comment:
I think this is meant to be internal.

Reply:
Ah OK I’ll remove for now. This was left over from the compiler changes